src/FSharp.Control.TaskSeq.Test/TaskSeq.SkipWhile.Tests.fs

```diff
@@ -0,0 +1,315 @@
+        // T T T T F F F F F F (stops at first F)
+        // x x x x x _ _ _ _ _ (skips inclusively)
+        // A B C D E F G H I J
+
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhileInclusive (fun x -> x < 5)
+            |> verifyDigitsAsString "FGHIJ" // last 4
+
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhileInclusiveAsync (fun x -> task { return x < 5 })
+            |> verifyDigitsAsString "FGHIJ"
+    }
+
+
+    [<Theory; ClassData(typeof<TestImmTaskSeq>)>]
+    let ``TaskSeq-skipWhileInclusive+A returns the empty sequence if always true`` variant = task {
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhileInclusive (fun x -> x > -1) // always true
+            |> verifyEmpty
+
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhileInclusiveAsync (fun x -> Task.fromResult (x > -1))
+            |> verifyEmpty
+    }
+
+    [<Theory; ClassData(typeof<TestImmTaskSeq>)>]
+    let ``TaskSeq-skipWhileInclusive+A always skips at least the first item`` variant = task {
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhileInclusive ((=) 0)
+            |> verifyDigitsAsString "BCDEFGHIJ"
+
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhileInclusiveAsync ((=) 0 >> Task.fromResult)
+            |> verifyDigitsAsString "BCDEFGHIJ"
+    }
+
+module SideEffects =
+    [<Theory; ClassData(typeof<TestSideEffectTaskSeq>)>]
+    let ``TaskSeq-skipWhile+A filters correctly`` variant = task {
+        // truth table for f(x) = x < 6
+        // 1 2 3 4 5 6 7 8 9 10
+        // T T T T T F F F F F (stops at first F)
+        // x x x x x _ _ _ _ _ (skips exclusively)
+        // A B C D E F G H I J
+
+        do!
+            Gen.getSeqWithSideEffect variant
+            |> TaskSeq.skipWhile (fun x -> x < 6)
+            |> verifyDigitsAsString "FGHIJ"
+
+        do!
+            Gen.getSeqWithSideEffect variant
+            |> TaskSeq.skipWhileAsync (fun x -> task { return x < 6 })
+            |> verifyDigitsAsString "FGHIJ"
+    }
+
+    [<Theory; ClassData(typeof<TestSideEffectTaskSeq>)>]
+    let ``TaskSeq-skipWhileInclusive+A filters correctly`` variant = task {
+        // truth table for f(x) = x < 6
+        // 1 2 3 4 5 6 7 8 9 10
+        // T T T T T F F F F F (stops at first F)
+        // x x x x x x _ _ _ _ (skips inclusively)
+        // A B C D E F G H I J
+
+        do!
+            Gen.getSeqWithSideEffect variant
+            |> TaskSeq.skipWhileInclusive (fun x -> x < 6)
+            |> verifyDigitsAsString "GHIJ"
+
+        do!
+            Gen.getSeqWithSideEffect variant
+            |> TaskSeq.skipWhileInclusiveAsync (fun x -> task { return x < 6 })
+            |> verifyDigitsAsString "GHIJ"
+    }
+
+    [<Fact>]
+    let ``TaskSeq-skipWhile and variants prove it reads the entire input stream`` () = task {
```

Comment:
see this warn? (IIRC you need to un-nest)

Reply:
@bartelink what warning? I don't have a warning...

Reply:
Oh shoot (GH doesn't show it on the main page, I should enable `warnings-as-error`).

> This state machine is not statically compilable. A constrained generic construct occured in the resumable code specification. An alternative dynamic implementation will be used, which may be slower. Consider adjusting your code to ensure this state machine is statically compilable, or else suppress this warning.

Reply:
this warning is not raised locally...

Reply:
should be resolved now, thanks for catching!

Reply:
yeah its releae mode only
I ran into one in Equinox so have a thing I would like to simplify but dont know how 
https://github.com/jet/equinox/blob/master/src/Equinox/Stream.fs#L33